hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -92,7 +92,6 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
                                Option<InternalSchema> internalSchemaOpt,
                                HoodieTableMetaClient hoodieTableMetaClient,
                                TypedProperties props,
-                               HoodieTableConfig tableConfig,
```

Comment:
Could you think about simplifying all the parameters?  The goal should be that given the file group ID and the query type with the reader context, storage instance, and the minimal set of configs (maybe with meta client), the file group reader should be able to figure out all necessary configs to fill in for reading the records out.

Reply:
Yeah, I can keep getting rid of more. Just want to keep the prs small

Reply:
+1